atom-client/src/main/java/org/motechproject/atomclient/service/impl/FeedCache.java

```diff
@@ -0,0 +1,290 @@
+package org.motechproject.atomclient.service.impl;
+
+import com.rometools.fetcher.impl.FeedFetcherCache;
+import com.rometools.fetcher.impl.SyndFeedInfo;
+import com.rometools.rome.feed.synd.SyndContent;
+import com.rometools.rome.feed.synd.SyndEntry;
+import com.rometools.rome.feed.synd.SyndFeed;
+import com.rometools.rome.io.FeedException;
+import com.rometools.rome.io.SyndFeedInput;
+import com.rometools.rome.io.SyndFeedOutput;
+import org.apache.commons.lang.ObjectUtils;
+import org.apache.commons.lang.StringUtils;
+import org.motechproject.atomclient.domain.FeedRecord;
+import org.motechproject.atomclient.repository.FeedRecordDataService;
+import org.motechproject.atomclient.service.AtomClientConfigService;
+import org.motechproject.atomclient.service.Constants;
+import org.motechproject.event.MotechEvent;
+import org.motechproject.event.listener.EventRelay;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.StringReader;
+import java.io.UnsupportedEncodingException;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class FeedCache implements FeedFetcherCache {
+
+
+    public static final String FEED_DATA = "feedData";
+    private FeedRecordDataService feedRecordDataService;
+    private AtomClientConfigService atomClientConfigService;
+    private EventRelay eventRelay;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(FeedCache.class);
+
+
+    public FeedCache(FeedRecordDataService feedRecordDataService, EventRelay eventRelay,
+                     AtomClientConfigService atomClientConfigService) {
+        this.feedRecordDataService = feedRecordDataService;
+        this.eventRelay = eventRelay;
+        this.atomClientConfigService = atomClientConfigService;
+    }
+
+
+    public static String feedToString(SyndFeed feed) throws FeedException, UnsupportedEncodingException {
```

Comment:
Most of those methods look like some internal logic of this class. Shouldn't they be private?